### PR TITLE
test(deployer): fixed deployer tests and removed unnecessary runtime logger_uri arg

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

Fixed deployer tests by implementing mocked logger clients based on mocked shuttle_proto::logger::Logger trait impls.

## How has this been tested? (if applicable)

Ran deployer tests locally and all passed.

### CODE REVIEW

File: cargo-shuttle/src/lib.rs [hunk 1]
```diff
@@ -861,7 +859,7 @@ impl Shuttle {
     async fn local_run(&self, run_args: RunArgs) -> Result<()> {
         let services = self.pre_local_run(&run_args).await?;
         let (provisioner_server, provisioner_port) = Shuttle::setup_local_provisioner().await?;
-        let (logger_server, logger_port) = Shuttle::setup_local_logger().await?;
+        let (logger_server, _) = Shuttle::setup_local_logger().await?;
```
Comment (chesedo) [inline cargo-shuttle/src/lib.rs:862]: I think we can complete remove the local logger now

File: codegen/src/shuttle_main/mod.rs
Comment (chesedo) [inline codegen/src/shuttle_main/mod.rs:1]: :partying_face:

File: deployer/Cargo.toml [hunk 2]
```diff
@@ -72,4 +72,6 @@ features = ["builder"]
 ctor = { workspace = true }
 hex = "0.4.3"
 rand = { workspace = true }
+shuttle-common-tests = { path = "../common-tests" }
```
Comment (chesedo) [inline deployer/Cargo.toml:75]: Maybe it should come from the workspace instead

**Review by chesedo — COMMENTED** [review]
Looks good. This also works :smile:

File: cargo-shuttle/src/lib.rs (hunk 1 above)
Comment (iulianbarbu) [reply cargo-shuttle/src/lib.rs]: Should be covered here: [ed97c57](https://github.com/shuttle-hq/shuttle/pull/1204/commits/ed97c5706c664278afc8b3e06af5700c57a1e472).

File: deployer/Cargo.toml (hunk 2 above)
Comment (iulianbarbu) [reply deployer/Cargo.toml]: Yes, should be better thanks: [b3eab46](https://github.com/shuttle-hq/shuttle/pull/1204/commits/b3eab46e0dd5a7e3d3ceb31ed401a2ba6716ca0f).

**Pull request merged**